app/src/main/java/com/arctouch/codechallenge/dashboard/DashboardActivity.kt

```diff
@@ -0,0 +1,77 @@
+package com.arctouch.codechallenge.dashboard
+
+import android.arch.lifecycle.Observer
+import android.arch.lifecycle.ViewModelProviders
+import android.content.Intent
+import android.os.Bundle
+import android.support.v7.app.AppCompatActivity
+import android.util.DisplayMetrics
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.dashboard.movielist.MovieListFragment
+import com.arctouch.codechallenge.details.DetailsActivity
+import com.arctouch.codechallenge.details.detailsFragment.DetailsFragment
+import com.arctouch.codechallenge.util.constants.IntentConstants
+import com.arctouch.codechallenge.viewmodels.MovieViewModel
+
+class DashboardActivity: AppCompatActivity() {
+
+    private var isPhone: Boolean = false
+    var viewModel: MovieViewModel? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.dashboard_activity)
+        setupScreenSize()
+        viewModel = ViewModelProviders.of(this).get(MovieViewModel::class.java)
+        setupView()
+        registerObservers()
+    }
+
+    //region Private
+
+    private fun setupView() {
+        supportActionBar?.setTitle(R.string.upcoming)
+        setupMovieListFragment()
+    }
+
+    private fun setupMovieListFragment() {
+        if (supportFragmentManager.fragments.isEmpty()) {
```

Comment:
When there is an orientation change the system automatically recreates fragments. Due to that, if that check was not made. The fragments would stack one above the other. Another way to solve this without the if would simply be to use .replace instead of the .add method.